src/RolloutScorer/RolloutScorerAzureFunction/RolloutScorerFunction.cs

```diff
@@ -1,135 +1,187 @@
 
             if (relevantDeployments.Count() > 0)
             {
-                log.LogInformation("INFO: Checking to see if the most recent deployment occurred more than two days ago...");
+                _logger.LogInformation("INFO: Checking to see if the most recent deployment occurred more than two days ago...");
                 // We have only want to score if the buffer period has elapsed since the last deployment
                 if ((relevantDeployments.Last().Ended ?? DateTimeOffset.MaxValue) < DateTimeOffset.UtcNow - TimeSpan.FromDays(ScoringBufferInDays))
                 {
                     var scorecards = new List<Scorecard>();
 
-                    log.LogInformation("INFO: Rollouts will be scored. Fetching GitHub PAT...");
-                    SecretBundle githubPat;
-                    using (KeyVaultClient kv = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(tokenProvider.KeyVaultTokenCallback)))
-                    {
-                        githubPat = await kv.GetSecretAsync(Utilities.KeyVaultUri, Utilities.GitHubPatSecretName);
-                    }
-
+                    _logger.LogInformation("INFO: Rollouts will be scored. Fetching GitHub PAT...");
+                    SecretBundle githubPat = await _keyVaultClient.GetSecretAsync(_keyVaultOptions.Uri, _gitHubOptions.PatSecretName);
+                    
                     // We'll score the deployments by service
                     foreach (var deploymentGroup in relevantDeployments.GroupBy(d => d.Service))
                     {
-                        log.LogInformation($"INFO: Scoring {deploymentGroup?.Count() ?? -1} rollouts for repo '{deploymentGroup.Key}'");
+                        string currentRepo = deploymentGroup.Key;
+
+                        _logger.LogInformation($"INFO: Scoring {deploymentGroup?.Count() ?? -1} rollouts for repo '{deploymentGroup.Key}'");
+
+                        _logger.LogInformation($"INFO: Finding repo config for {currentRepo}...");
+                        RepoConfig currentRepoConfig = Configs.DefaultConfig.RepoConfigs[currentRepo];
+                        _logger.LogInformation($"INFO: Repo config: {currentRepoConfig.Repo}");
+
+                        _logger.LogInformation($"INFO: Finding AzDO config for {currentRepoConfig.AzdoInstance}...");
+                        AzdoInstanceConfig currentAzdoInstanceConfig = Configs.DefaultConfig.AzdoInstanceConfigs[currentRepoConfig.AzdoInstance];
+
                         RolloutScorer.RolloutScorer rolloutScorer = new RolloutScorer.RolloutScorer
                         {
                             Repo = deploymentGroup.Key,
-                            RolloutStartDate = deploymentGroup.First().Started.GetValueOrDefault().Date,
+                            RolloutStartDate = deploymentGroup.First().Started.GetValueOrDefault().Date,  // Is this going to blow up if GetValueOrDefault returns null?
```

Comment:
Yes, yes it will. :-)

Reply:
Wait, no GetValueOrDefault on a Nullable can't return null... this is a weird line... because the "First" might crash.  It will just return "1/1/0000", which isn't useful... I feel like the Nullable crash would be better than it proceeding with illegal defaults.

"GetValueOrDefault" is usually a bad sign.

Reply:
Or mabye that's fine because it's the start? Don't know. :-D

Reply:
null check on First() and started should help before GetValue

Reply:
You'll have to switch it to FirstOrDefault(), since First just throws. ;-)